developers/weaviate/concepts/indexing/dynamic-indexes.mdx

```diff
@@ -0,0 +1,50 @@
+---
+title: Dynamic indexes
+sidebar_position: 40
+image: og/docs/indexing.jpg
+# tags: ['basics']['indexes']
+---
+
+import DynamicIntro from '/_includes/indexes/dynamic-intro.mdx';
+
+<DynamicIntro/>
+
+To configure a dynamic index, see: [Configure dynamic indexes](/developers/weaviate/configuration/indexing-vector/dynamic-indexes).
+
+## Overview
+
+Dynamic indexes are flat indexed collections that Weaviate converts to HNSW indexed collections when the collection reaches a certain size. Flat indexes work well for collections with less than 10,000 objects. At that size, flat indexes have low memory overhead and good latency. But, search latency increases as the number of objects in a collection increases. When the collection grows to about 10,000 objects, an HNSW index usually has better latency than a flat index.
+
+The dynamic index helps to balance resource costs against search latency times. Flat indexes are disc-based. They are responsive at low object counts, but get slower as object counts grow. HNSW indexes reside in RAM. They are very fast, but RAM is expensive. Disk storage is orders of magnitude cheaper than RAM memory, so hosting an index on disc is significantly cheaper than hosting it in RAM.
+
+If your collection size grows over time, or if you have a mix of smaller and larger tenants, dynamic indexes let you take advantage of lower cost flat indexes while object counts and search latency times are low. When the object count increase, and latencies grow larger, converting the flat index to an HNSW index preserves low search latencies at the expense of increased RAM memory costs.
```

Comment:
> If your collection size grows over time, or if you have a mix of smaller and larger tenants, 

This kind of comes out suddently without discussing multi-tenancy, and might be a bit jarring for the reader.